src/AutoRest.CSharp/Mgmt/Decorator/Transformer/AbstractTypeDetection.cs

```diff
@@ -0,0 +1,65 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Text;
+using AutoRest.CSharp.Input;
+using AutoRest.CSharp.Mgmt.AutoRest;
+using AutoRest.CSharp.Mgmt.Output;
+using AutoRest.CSharp.Output.Models.Types;
+using Microsoft.CodeAnalysis;
+
+namespace AutoRest.CSharp.Mgmt.Decorator.Transformer
+{
+    internal static class AbstractTypeDetection
+    {
+        public static readonly ConcurrentDictionary<string, ObjectSchema> AbstractTypeToInternalDerivedClass = new();
```

Comment:
It isn't best to use the original schema name as the key to the dictionary, it's possible we couldn't find the corresponding value when the name is updated after this detection, need an update here.

Reply:
Remove this change.